pkg/trace/api/api.go

```diff
@@ -410,51 +422,35 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
 		for _, b := range group.Stats {
 			// Normalizer
 			b.Name, _ = traceutil.NormalizeName(b.Name)
-			if b.Env == "" {
-				b.Env = r.conf.DefaultEnv
-			}
-			b.Env = traceutil.NormalizeTag(b.Env)
 			b.Service, _ = traceutil.NormalizeService(b.Service, req.Header.Get(headerLang))
 			if b.Resource == "" {
 				b.Resource = b.Name
 			}
-			// TODO(x): obfuscate b.Resource
 			b.Resource, _ = traceutil.TruncateResource(b.Resource)
-			// TODO(x): validate b.HTTPStatusCode if added
-
-			tags := map[string]string{
-				"version": b.Version,
-			}
-			for _, t := range b.OtherTags {
-				parts := strings.SplitN(t, ":", 2)
-				if len(parts) > 0 {
-					if v := parts[0]; len(v) > traceutil.MaxMetaKeyLen {
-						parts[0] = traceutil.TruncateUTF8(v, traceutil.MaxMetaKeyLen) + "..."
-					}
-					tags[parts[0]] = ""
-				}
-				if len(parts) > 1 {
-					if v := parts[1]; len(v) > traceutil.MaxMetaValLen {
-						parts[1] = traceutil.TruncateUTF8(v, traceutil.MaxMetaValLen) + "..."
-					}
-					tags[parts[0]] = parts[1]
+			if b.Type == "sql" {
+				// TODO(gbbr): perhaps we should store only one instance instead of
+				// a new one on each request?
+				oq, err := obfuscate.NewObfuscator(nil).ObfuscateSQLString(b.Resource)
+				if err != nil {
+					log.Errorf("Error obfuscating resource %q: %v", b.Resource, err)
+				} else {
+					b.Resource = oq.Query
 				}
 			}
 
+			tags := map[string]string{"version": in.Version}
+			if b.HTTPStatusCode != 0 {
+				tags["http.status_code"] = strconv.Itoa(int(b.HTTPStatusCode))
```

Comment:
Note that [this](https://github.com/DataDog/datadog-agent/blob/18565f2c5cf9c28d4f52ed0b6930384b082f3972/pkg/trace/agent/normalizer.go#L162) was the previous validation happening on the HTTP status code. I think there's no need for it now that we're an `int`.